### PR TITLE
164: a follow-up PR of PR  #164 hook up data point graphing to the dal

### LINKED ISSUE #164: Hook up the data-point reading to the DAL
Right now our data points are still statically declared. We ned to hook that up to read data from the DAL

### PR DESCRIPTION
**Pull request by LongQuanNguyen**
Closes #164 

# What was the issue
Some clean-up tasks after the main PR in 164 is merged:
- Delete unused files from the spike
- Change @preconds to @pre-conditions
- Refactor DAL's storeCsv method and add more tests
- Add missing documents

### CODE REVIEW

File: my-webxr-app/tests/data/DataLayer.test.tsx [hunk 1]
```diff
@@ -195,6 +195,54 @@ describe('Validate storeCSV() operation', () => {
...
+    const resolve = await dataLayer.storeCSV(batchItems);
+    expect(resolve).toEqual(true);
+
+    const column1 = await repository.getColumn('column1', TableName.RAW);
+    const column2 = await repository.getColumn('column2', TableName.RAW);
+
+    expect(column1.values).toHaveLength(0);
+    expect(column2.values).toHaveLength(0);
+  });
+
```
Comment (Aesha7) [inline my-webxr-app/tests/data/DataLayer.test.tsx:245]: Great tests!

Do we need to test if batch data stream entires is null?

File: my-webxr-app/src/data/DataLayer.tsx [hunk 2]
```diff
@@ -68,39 +67,49 @@ export default class DataLayer implements DataAbstractor {
...
+          const newColumn = new Column<NumericColumn>(columnName, []);
+          await this.repository.addColumn(newColumn, TableName.RAW);
+        });
+        await Promise.all(promises);
+
+        // Remove the header row (first row) from the batchItems
+        if (batchItems.length >= 1) {
+          /* eslint-disable no-param-reassign */
+          batchItems = batchItems.slice(1);
+        }
```
Comment (Aesha7) [inline my-webxr-app/src/data/DataLayer.tsx:95]: This seems good to me

File: my-webxr-app/src/repository/DbRepository.tsx [hunk 3]
```diff
@@ -205,7 +205,7 @@ export default class DbRepository extends Dexie implements Repository {
    * Retrieves the x, y, and z values from the specified columns in the database and returns them as
    * an array of DataPoint objects.
    *
-   * @preconds
+   * @pre-conditions
```
Comment (Aesha7) [inline my-webxr-app/src/repository/DbRepository.tsx:208]: Great clean up with documentation 🙌

**Review by Aesha7 — APPROVED** [review]

File: my-webxr-app/tests/data/DataLayer.test.tsx (hunk 1 above)
Comment (trangnguyen3010) [reply my-webxr-app/tests/data/DataLayer.test.tsx:245]: I don't think we need to since in Typescript. Null and Undefined aren't a value of an object but are rather a type. Since we already restrict the type of argument to BatchedDataStream (which can only be wither Array[string] or Array[number], we're already sure that the variable must be of type BatchedDataStream (not any other type like Null or Undefined) ?

Comment (LongQuanNguyen) [reply my-webxr-app/tests/data/DataLayer.test.tsx:245]: TypeScript is weird ngl. I love the idea, but still hate it. As Trang's comment, we don't need to check it, given TypeScript does it job

File: my-webxr-app/src/data/DataLayer.tsx [hunk 4]
```diff
@@ -68,39 +67,49 @@ export default class DataLayer implements DataAbstractor {
...
+      }
+
+      if (batchItems.length > 0) {
+        const transposedData = DataLayer.transposeData(batchItems);
+        const columnNames = await this.repository.getCsvColumnNames();
+        const promises = transposedData.map(async (column, index) => {
+          const columnName = columnNames[index];
+          const newValues = column;
+          // push new values to the column in the repository
```
Comment (trangnguyen3010) [inline my-webxr-app/src/data/DataLayer.tsx:104]: would this cause too much of a merge conflict with your PR @Jesster2829 @LongQuanNguyen ? if it does, I'll revert it

Comment (LongQuanNguyen) [reply my-webxr-app/src/data/DataLayer.tsx:104]: It may, I am not sure how much it will be. May be wait until Jesse's PR get merged in first

**Comment by trangnguyen3010** [comment]
Not important and too much merge conflict to handle
